### PR TITLE
Add regression tests of inference of implicit None return

### PR DESCRIPTION
**Pull request by nelfin**
## Steps

- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [ ] ~Add a ChangeLog entry describing what your PR does.~
- [ ] ~If it's a new feature or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.~
- [x] Write a good description on what the PR does.

## Description

These errors are not currently raised on `pylint 2.8.x` and `astroid 2.5.6`

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Related Issue

Ref PyCQA/astroid#485. Depends on PyCQA/astroid#983.

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
Thank you, that's pretty clean. I won't have to approve all your pipelines for much longer :wink:

File: tests/functional/u/unexpected_special_method_signature.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """Test for special methods implemented incorrectly."""
 
-# pylint: disable=missing-docstring, unused-argument, too-few-public-methods
+# pylint: disable=missing-docstring, unused-argument, too-few-public-methods, invalid-format-returned, non-iterator-returned
```
Comment (cdce8p) [inline tests/functional/u/unexpected_special_method_signature.py:3]:
```suggestion
# pylint: disable=missing-docstring, unused-argument, too-few-public-methods
```
These don't seem to be necessary anymore

**Comment by Pierre-Sassoulas** [comment]
This can be rebased on master to get astroid ``2.5.7``.

**Comment by cdce8p** [comment]
Part has been merged with #4523: https://github.com/PyCQA/pylint/commit/6f1ccd8ed25eebb30a4b499c46c8b5f041f50fac
